theano/tensor/nnet/corr.py

```diff
@@ -271,13 +283,13 @@ def c_code_helper(self, bottom, weights, top, sub, height=None, width=None):
         if height:
             height = '(*(npy_int64 *)(PyArray_DATA(%s)))' % height
         else:
-            if ((self.direction != 0) and (self.dH != 1)) or ((self.direction == 1) and (self.padH == -1)):
+            if ((self.direction != 0) and (self.dH != 1)) or ((self.direction == 1) and (self.padH_l == -1)):
```

Comment:
Maybe check for `self.padH_r == -1` as well (you do that in all the instances below).